app/routes/search.tsx

```diff
@@ -1,194 +1,52 @@
-import type { LoaderFunction } from '@remix-run/node';
 import { useLoaderData } from '@remix-run/react';
-import { getEvents, type Event } from '~/models/events.server';
+import type { LoaderFunction } from '@remix-run/node';
+import { db } from '~/modules/database/db.server';
+import { EventsCards } from '~/components/marketing/events-section';
+import type { Event } from '@prisma/client';
 
 export const loader: LoaderFunction = async ({ request }) => {
-  // Get the query string from the URL
-
   const url = new URL(request.url);
   const query = url.searchParams.get('q');
 
-  // Only numbers, letters are allowed
-
-  const regex = /^[a-zA-Z0-9]+$/;
+  // Some validation. Can show error message with Remix instead the Response.
 
-  // Some validation
-
-  if (typeof query !== 'string' || !query || query.length > 50 || !regex.test(query)) {
+  if (typeof query !== 'string' || !query || query.length > 50) {
```

Comment:
As previously discussed with @tonydangblog, adding Regular Expression would be "overkill" for this case.

Reply:
Looks good for now!